src/components/Map.vue

```diff
@@ -1,9 +1,9 @@
 <template>
     <div id="app">
-        <search-bar ref="search"></search-bar>
-        <scroll-bar ref="scroll"></scroll-bar>
-        <map-view-button :map="this.map"></map-view-button>
-        <div id="map" style="height: 100vh"></div>
+        <div id="map" style="height: 100vh">
+            <search-bar :route="route" ref="search"></search-bar>
```

Comment:
Careful your merge got rid of code in master that we need. Like the map view button, and adding map actions again. Make sure that when you merge from master you understand the code that you are merging so that we do not go back to a previous state of master.  One way to check this is to see how master looks, and see if after you merge all the changes are still there. Make sure the 2D button visible again and get rid of mapAction, we do not need it here.  The search bar does not need to have this action passed to it.